### PR TITLE
chore: setup analyzer plugin as typescript project FUI-1528

### PR DESCRIPTION
**Pull request by matteematt**
🤔  &nbsp; **What does this PR do?**

- Implements typescript in the analyzer plugin module
- Adds basic unit testing for analyzer plugin module
- This does *not* add in the actual analyzer plugin we need - that will be part of a subsequent PR!
- Bumps version of eslint prettier plugin which created some linting issues
- Fixes the linting issues in the lsp which arose from the above change (basically just adding loads of trailing commas)

📑  &nbsp; **How should this be tested?**

Update test steps to match your PR:

```
1. Checkout branch
2. `npm run bootstrap`
3. `npm run test`

To test the LSP is working correctly in VSCode
1. `cd packages/showcase/example`
2. `code .` (or manually open VSCode via the UI to the `example` directory on your filesystem).
3. Open a typescript file such as `root.ts`
4. In the Command Palette choose `TypeScript: select typescript version...` and then choose the workspace version
5. The LSP should be enabled. You will need to make a code change after the LSP is enabled before you see any Intellisense.
```

✅  &nbsp; **Checklist**

- [X] I have added tests for my changes.
- [X] I have updated the project documentation.
- [X] I have followed the [contribution guidelines](https://github.com/genesiscommunitysuccess/custom-elements-lsp/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: packages/core/custom-elements-lsp/jest/custom-elements.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ let manifest: string;
  * Builds a stubbed `CustomElementsResource` and constructs a `CustomElementsServiceImpl`.
```
Comment (matteematt) [inline packages/core/custom-elements-lsp/jest/custom-elements.ts:13]: All of the changes in `core/custom-elements-lsp` are just linting changes arising from needing the bump the eslint prettier plugin

File: packages/core/analyzer-import-alias-plugin/src/index.ts [hunk 2]
```diff
@@ -1,25 +1,40 @@
+import { Plugin } from '@custom-elements-manifest/analyzer';
```
Comment (matteematt) [inline packages/core/analyzer-import-alias-plugin/src/index.ts:1]: This isn't actually the plugin code we need - that will be addressed in a subsequent PR. This PR is just implementing the minimal required change to convert this existing file to typescript (that is why there is such liberal use of `any`).

File: .github/workflows/CI_merge.yml [hunk 3]
```diff
@@ -29,8 +29,8 @@ jobs:
       - name: Lint
         run: npm run lint:eslint:all
 
-      - name: Unit Test
-        run: npm run test:unit
+      - name: Tests
```
Comment (SzymonZur) [inline .github/workflows/CI_merge.yml:32]: not a blocker, just a nitpick, but shouldn't it be consistent with CI_action_publish_master.yml and named "Test"?

File: .github/workflows/CI_pr.yml [hunk 4]
```diff
@@ -26,5 +26,5 @@ jobs:
       - name: Lint
         run: npm run lint:eslint:all
 
-      - name: Unit Test
-        run: npm run test:unit
+      - name: Tests
```
Comment (SzymonZur) [inline .github/workflows/CI_pr.yml:29]: Same here

**Review by SzymonZur — APPROVED** [review]

**Comment by matteematt** [comment]
@SzymonZur thanks for the review. You are correct, though they're small issues so will bring those changes into the PR I am currently working on

**Pull request merged**